### PR TITLE
AIRO-1368 Allow building debian package 

### PR DESCRIPTION
**Pull request by RafaelRey**
## Proposed change(s)

This change includes all necessary changes to allow creating a debian .deb package with tools like python-bloom and run the inside python scripts with and without a python virtual environment. 

- Debian maintainer script files: A postinst and postrm has been added to allow creating and removing the python virtual environment in the ros package path using the dependencies listed in the requirements.txt
- CMakeLists: Added installs commandas for python programs, launch, config and requirements.txt files
- endpoint.launch: Added argument to allow users choose to run the package with or without the virtual environment. Useful for users that download the source and run it without a virtual environment. *false by default* 
- package.xml: Added extra execution depends
- requirements.txt
### Useful links 

Related github issue:
#99 

### Types of change(s)

- [ ] Bug fix
- [x] New feature
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other (please describe)

## Testing and Verification

Please describe the tests that you ran to verify your changes. Please also provide instructions, ROS packages, and Unity project files as appropriate so we can reproduce the test environment.

The debian has been generated, installed, updated, removed and purged with the expected behavior. It has also been ran with and without the virtual environment

### Test Configuration:

- ROS machine OS + version: [Ubuntu 20.04, ROS Noetic]
Debian package created with the help of aun auxiliary script  [generate_debian_pkgs.sh](https://github.com/robotics-upo/rosdistro/blob/master/generate_debian_pkgs.sh) this helper script only adds some features to standard ros package python-bloom
Clone the package in the src/ folder of a clean workspace and:

```bash
cd <CLEAN_WS>
wget https://raw.githubusercontent.com/robotics-upo/rosdistro/master/generate_debian_pkgs.sh
sudo chmod +x generate_debian_pkgs.sh
./generate_debian_pkgs.sh --output_folder=$(pwd) --workspace_folder=$(pwd)
```

Now the debian should be created in the workspace root. To install it:
```bash
sudo dpkg -i ros-noetic-ros-tcp-endpoint_0.6.0-0focal_amd64.deb
```

To test it make sure you have not sourced your catkin workspace where the package is also downloaded (It has preference over the installed version). So if its sourced, unsource it (maybe you need also to run ```rospack profile``` to detect new installed package) and check that everything is working for both modes, with and without virtual environment:

```bash
roslaunch ros_tcp_endpoint endpoint.launch enable_py_venv:=true
```
or without python virtual environment (default behavior, as it is the current one in the dev branch)

```bash
roslaunch ros_tcp_endpoint endpoint.launch 
```

## Checklist
- [x] Ensured this PR is up-to-date with the `dev` branch
- [x] Created this PR to target the `dev` branch
- [x] Followed the style guidelines as described in the [Contribution Guidelines](https://github.com/Unity-Technologies/ROS-TCP-Endpoint/blob/main/CONTRIBUTING.md)
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [Changelog](https://github.com/Unity-Technologies/ROS-TCP-Endpoint/blob/dev/CHANGELOG.md) and described changes in the [Unreleased section](https://github.com/Unity-Technologies/ROS-TCP-Endpoint/blob/dev/CHANGELOG.md#unreleased)
- [ ] Updated the documentation as appropriate
- [x] Debian maintainer scripts (postinst, postrm) verified with [shellcheck](https://www.shellcheck.net/)
## Other comments

### CODE REVIEW

File: src/ros_tcp_endpoint/default_server_endpoint_venv.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import os
+import subprocess
+import sys
+from time import sleep
+
+def is_venv():
+    return (hasattr(sys, 'real_prefix') or
+            (hasattr(sys, 'base_prefix') and sys.base_prefix != sys.prefix))
+
+arg='--venv'
```
Comment (sdiao) [inline src/ros_tcp_endpoint/default_server_endpoint_venv.py:12]: nit: Could you fix the formatting in this PR? The [pre-commi](https://github.com/Unity-Technologies/ROS-TCP-Endpoint/runs/4217867319?check_suite_focus=true)t github action has a list of formatting suggestions.

**Comment by mrpropellers** [comment]
Hi @RafaelRey - thanks for doing this work and the thorough testing instructions! I've set aside some time tomorrow to pull down your changes and verify that I see the same behavior you've described in your readme.

**Review by mrpropellers — CHANGES_REQUESTED** [review]
@RafaelRey So I was able to get it to install with a little massaging. I ran out of time to do a full end-to-end test, but did get the endpoint to fully spin up from the .deb install. **I use a fairly streamlined docker image built from the noetic base image so I had to do a bit of work**:
* Needed to install these dependencies
  * python-bloom
  * fakeroot
  * debhelper
  * python3-pip
  * python3-venv
  * virtualenv
* Had to run `catkin_make` inside my clean workspace before generate_debian_pkgs.sh would recognize it as a valid catkin workspace
* Needed to run `rosdep update` to fix the following error during packaging:
```
ValidationFailed: rosdep database does not have any sources.
Please make sure you have a valid configuration in:
        /etc/ros/rosdep/sources.list.d
```
Other than that, though, seems to work great! I'd like to request the following changes: 
1. Can you update the instructions to include steps for installing the above dependencies and running the two commands to ensure the workspace is ready to be packaged
2. Could you move or copy these instructions from the PR into their own .md file and add a section to the repo's README.md, under **Introduction**, titled "Building Into a Debian Package" with the requested updates?
3. Double-check the pre-commit output and ensure your formatting follows the recommendations there. Note that it's picky about line endings and tabs/spaces issues.

I will do an end-to-end test in the morning but assuming I don't find anything, this should be otherwise ready to be merged in.

**Comment by mrpropellers** [comment]
Hi @RafaelRey, have you had a chance to look at the changes requested here?

**Comment by mrpropellers** [comment]
Closing as this is now very stale. Assuming the user has a version they're happy with on their fork.

**Pull request closed without merging**